builtin/game/item.lua

```diff
@@ -676,9 +676,7 @@ function core.node_dig(pos, node, digger)
 	-- Run script hook
 	for _, callback in ipairs(core.registered_on_dignodes) do
 		local origin = core.callback_origins[callback]
-		if origin then
-			core.set_last_run_mod(origin.mod)
-		end
+		core.set_last_run_mod(origin.mod or "??")
```

Comment:
this should be nil, the `"??"` fallback is generated at the point of use

Reply:
> this should be nil, the `"??"` fallback is generated at the point of use

But does `core.set_last_run_mod` accept `nil`? I believe it requires the argument to be a `string`; only later on does the C++ func fallback to `??` if the `char*` is a `nullptr`.

Reply:
Indeed it only takes strings, I suggest `""` then plus this change:
```diff
diff --git a/src/script/common/c_internal.cpp b/src/script/common/c_internal.cpp
index df82dba14..827cce590 100644
--- a/src/script/common/c_internal.cpp
+++ b/src/script/common/c_internal.cpp
@@ -76,10 +76,10 @@ void script_error(lua_State *L, int pcall_result, const char *mod, const char *f
                err_type = "Unknown";
        }
 
-       if (!mod)
+       if (!mod || !*mod)
                mod = "??";
 
-       if (!fxn)
+       if (!fxn || !*fxn)
                fxn = "??";
 
        const char *err_descr = lua_tostring(L, -1);
```

Reply:
Looks like `"??"` is set as a default in numerous places:

```
$ grep -r '"??"'
builtin/profiler/instrumentation.lua:	def.mod = def.mod or get_current_modname() or "??"
builtin/common/chatcommands.lua:	def.mod_origin = core.get_current_modname() or "??"
builtin/client/register.lua:			mod = core.get_current_modname() or "??",
builtin/client/register.lua:			name = getinfo(1, "n").name or "??"
builtin/game/item.lua:		core.set_last_run_mod(origin.mod or "??")
builtin/game/register.lua:	spec.mod_origin = core.get_current_modname() or "??"
builtin/game/register.lua:	spec.mod_origin = core.get_current_modname() or "??"
builtin/game/register.lua:	prototype.mod_origin = core.get_current_modname() or "??"
builtin/game/register.lua:	itemdef.mod_origin = core.get_current_modname() or "??"
builtin/game/register.lua:		core.set_last_run_mod(origin.mod or "??")
builtin/game/register.lua:			mod = core.get_current_modname() or "??",
builtin/game/register.lua:			name = debug.getinfo(1, "n").name or "??"
builtin/game/register.lua:			mod = core.get_current_modname() or "??",
builtin/game/register.lua:			name = debug.getinfo(1, "n").name or "??"
builtin/game/register.lua:		mod = core.get_current_modname() or "??",
builtin/game/register.lua:		name = debug.getinfo(1, "n").name or "??"
builtin/game/detached_inventory.lua:	stuff.mod_origin = core.get_current_modname() or "??"
src/script/common/c_internal.cpp:		mod = "??";
src/script/common/c_internal.cpp:		fxn = "??";
src/script/cpp_api/s_base.cpp:	m_last_run_mod = origin ? origin : "??";
doc/lua_api.txt:        -- if it can not find a source, returns "??"
```